src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/compute_tke_dissipation_diffusivity_fields.jl

```diff
@@ -0,0 +1,218 @@
+#####
+##### Diffusivities and diffusivity fields utilities
+#####
+
+function DiffusivityFields(grid, tracer_names, bcs, closure::FlavorOfKEpsilon)
+
+    default_diffusivity_bcs = (κu = FieldBoundaryConditions(grid, (Center, Center, Face)),
+                               κc = FieldBoundaryConditions(grid, (Center, Center, Face)),
+                               κe = FieldBoundaryConditions(grid, (Center, Center, Face)),
+                               κϵ = FieldBoundaryConditions(grid, (Center, Center, Face)))
+
+    bcs = merge(default_diffusivity_bcs, bcs)
+
+    κu = ZFaceField(grid, boundary_conditions=bcs.κu)
+    κc = ZFaceField(grid, boundary_conditions=bcs.κc)
+    κe = ZFaceField(grid, boundary_conditions=bcs.κe)
+    κϵ = ZFaceField(grid, boundary_conditions=bcs.κϵ)
+    Le = CenterField(grid)
+    Lϵ = CenterField(grid)
+
+    # Secret tuple for getting tracer diffusivity_fields with tuple[tracer_index]
+    tracer_diffusivity_fields = Dict(name => κc for name in tracer_names)
+    tracer_diffusivity_fields[:e] = κe
+    tracer_diffusivity_fields[:ϵ] = κϵ
+
+    implicit_linear_coefficients = Dict{Symbol, Any}(name => ZeroField() for name in tracer_names)
+    implicit_linear_coefficients[:e] = Le
+    implicit_linear_coefficients[:ϵ] = Lϵ
+
+    _tupled_tracer_diffusivity_fields = NamedTuple(name => tracer_diffusivity_fields[name] for name in tracer_names)
+    _tupled_implicit_linear_coefficients = NamedTuple(name => implicit_linear_coefficients[name] for name in tracer_names)
+
+    return (; κu, κc, κe, κϵ, Le, Lϵ,
+            _tupled_tracer_diffusivity_fields,
+            _tupled_implicit_linear_coefficients)
+end        
+
+@inline viscosity(::FlavorOfKEpsilon, diffusivity_fields) = diffusivity_fields.κu
+@inline diffusivity(::FlavorOfKEpsilon, diffusivity_fields, ::Val{id}) where id =
+    diffusivity_fields._tupled_tracer_diffusivity_fields[id]
+
+function compute_diffusivities!(diffusivity_fields, closure::FlavorOfKEpsilon, model; parameters = :xyz)
```

Comment:
You changed a lot of the names from `diffusivities` to `diffusivity_fields`, which looks better to me.

In this spirit, did you want to call this `compute_diffusivity_fields`?  I'm find either way, just a suggestion.

Reply:
yes eventually! that is a bigger change so I didn't do it yet

Reply:
I am very excited to see this PR come to life.  Thanks for getting it started!